### DISCUSSION TITLE
Equivalent of latex wrapfig

### DISCUSSION THREAD

**Discussion by ethanabrooks** [typst/typst#3088] [General]
### Description

The latex [wrapfig](https://ctan.org/pkg/wrapfig?lang=en) package allows placement of a figure in the midst of text, with text wrapping around it. Is it possible to implement such a thing in typst?

Here is an example image:

![image](https://github.com/typst/typst/assets/10344742/a9beb651-ccdf-4c31-b289-3d24331d2948)

I am also just curious to learn more about how layouts are implemented in typst.

### Use Case

If a figure is much narrower than text, it looks awkward to allocate the full text-width for the figure.

**Comment by Dherse** [comment]
While it is definitely planned see [somewhere on the roadmap](https://typst.app/docs/roadmap/), it is not built-in yet since it depends on some reworks of the internals of Typst, but it is coming. My current workflow for it is to do it manually with grids in the middle of the document. Definitely not ideal, but it'll improve!
